### PR TITLE
[logging] Fix a few small regressions

### PR DESCRIPTION
**Pull request by olivielpeau**
### What does this PR do?

Addresses a couple of regressions introduced in https://github.com/DataDog/datadog-agent/pull/5671/:

1. Do not panic if the agent is configured to log nowhere (instead the agent should print an error and exit)
2. Fix the logging of log entries logged before logger init (they were not logged at all anymore)

\+ fixes a small unrelated issue with the additional stack depth of errors logged from the internal HTTP server
\+ adds a release note and documentation on the `jmx_log_file` option that that PR introduced

### Additional Notes

Each change is in its own commit.

### Describe your test plan

Unfortunately the logging logic is hard to unit test. I've done the following manual testing:
* `Do not panic if the agent is configured to log nowhere`: set the agent to log nowhere (`disable_file_logging: true` and `log_to_console: false`), the agent should exit with an error instead of crashing on a nil pointer.
* `Fix the logging of log entries logged before logger init (they were not logged at all anymore)`: put a non-existing option in `datadog.yaml` and start the agent: the agent should log `Unknown key in config file: [...]` at startup.
* `issue with the additional stack depth of errors logged from the internal HTTP server`: make the http server log an error (for example with `curl https://localhost:5001`), check the stack context in the logged error is relevant.

### CODE REVIEW

**Review by prognant — APPROVED** [review]
🙇‍♂️ LGTM

File: pkg/util/log/log.go [hunk 1]
```diff
@@ -44,6 +44,15 @@ type DatadogLogger struct {
...
 	logger = setupCommonLogger(i, level)
+
+	// Flush the log entries logged before initialization now that the logger is initialized
+	bufferMutex.Lock()
+	bufferLogsBeforeInit = false
+	defer bufferMutex.Unlock()
+	for _, logLine := range logsBuffer {
+		logLine()
+	}
+	logsBuffer = []func(){}
```
Comment (ogaca-dd) [inline pkg/util/log/log.go:55]: Why this code is not needed for `SetupJMXLogger`?

File: pkg/config/config_template.yaml [hunk 2]
```diff
@@ -1180,6 +1180,11 @@ api_key:
 #
 # jmx_use_container_support: false
 
+## @param jmx_log_file - string - optional
```
Comment (ogaca-dd) [inline pkg/config/config_template.yaml:1183]: What do you think about adding the default value, ie `- default: jmxfetch.log`?

**Review by ogaca-dd — APPROVED** [review]
LGTM

File: pkg/util/log/log.go (hunk 1 above)
Comment (olivielpeau) [reply pkg/util/log/log.go:55]: This logic allows actually logging the log messages that are submitted to the main agent logger before it's initialized (for example, log messages logged when the main config is initialized). This is irrelevant to the jmxfetch logger (the jmxfetch logger is not the default one, and is only used after it's initialized)

File: pkg/config/config_template.yaml (hunk 2 above)
Comment (olivielpeau) [reply pkg/config/config_template.yaml:1183]: The default depends on the OS and should be an absolute path, so it's tricky to document it here (and should not be documented as a relative path). The `log_file` config option links to this [docs page](https://docs.datadoghq.com/agent/guide/agent-log-files/?tab=agentv6v7) that we should update for 7.24 (in a future PR) to mention this new file.

**Pull request merged**